tools/ts_library_builder/build_library.ts

```diff
@@ -0,0 +1,450 @@
+import { writeFileSync } from "fs";
+import * as prettier from "prettier";
+import {
+  ExpressionStatement,
+  ModuleKind,
+  ModuleResolutionKind,
+  NamespaceDeclarationKind,
+  Project,
+  ScriptTarget,
+  SourceFile,
+  Type,
+  TypeGuards
+} from "ts-simple-ast";
+import {
+  addInterfaceProperty,
+  addSourceComment,
+  addVariableDeclaration,
+  checkDiagnostics,
+  flattenNamespace,
+  getSourceComment,
+  loadDtsFiles,
+  loadFiles,
+  namespaceSourceFile,
+  normalizeSlashes
+} from "./ast_util";
+
+export interface BuildLibraryOptions {
+  /**
+   * The path to the root of the deno repository
+   */
+  basePath: string;
+
+  /**
+   * The path to the current build path
+   */
+  buildPath: string;
+
+  /**
+   * Denotes if the library should be built with debug information (comments
+   * that indicate the source of the types)
+   */
+  debug?: boolean;
+
+  /**
+   * The path to the output library
+   */
+  outFile: string;
+
+  /**
+   * Execute in silent mode or not
+   */
+  silent?: boolean;
+}
+
+/**
+ * A preamble which is appended to the start of the library.
+ */
+// tslint:disable-next-line:max-line-length
+const libPreamble = `// Copyright 2018 the Deno authors. All rights reserved. MIT license.
+
+/// <reference no-default-lib="true" />
+/// <reference lib="esnext" />
+
+`;
+
+// The path to the msg_generated file relative to the build path
+const MSG_GENERATED_PATH = "/gen/msg_generated.ts";
+
+// An array of enums we want to expose pub
+const MSG_GENERATED_ENUMS = ["ErrorKind"];
+
+/** Extracts enums from a source file */
+function extract(sourceFile: SourceFile, enumNames: string[]): string {
+  // Copy specified enums from msg_generated
+  let output = "";
+  for (const enumName of enumNames) {
+    const enumDeclaration = sourceFile.getEnumOrThrow(enumName);
+    enumDeclaration.setHasDeclareKeyword(false);
+    output += enumDeclaration.getText();
```

Comment:
You probably already know this and not sure if it's necessary here, but just take note that `.getText()` does not include the js docs. If you want that, there is a `.getText(true)` argument that will include the js docs when getting the text (not a fan of obscure boolean parameters, but this patterns off `getStart(includeJsDocComments = false)` from the compiler api). 

Also note, `.getText()` returns the text without leading trivia (comments & whitespace) while `.getFullText()` returns the text with leading trivia.

Reply:
Specifically here the generated declaration file includes a useless generated JSDoc comment so it is intentional, but I should comment that.